### DISCUSSION TITLE
Evaluation of Survival Models

### DISCUSSION THREAD

**Discussion by VanishingRasengan** [sebp/scikit-survival#265] [Q&A] [answered]
Dear @sebp,

first of all thanks for creating this amazing Python module!
I have a question when it comes to the evaluation of survival models. In one of your notebooks you present the c-index and the "time-dependent area under the ROC" as some options. However, after reading several papers I have the feeling that there seems to be no common consensus on this field.
Particularly, I trained some survival models provided by scikit-survival on my dataset and want to show that e.g. Random Survival Forest is superior to the other ones when it comes to survival/prognosis prediction. Is it enough to compare the c-indices? I am quite confused because in literature you only find some confusion matrices or ROC-curves but that does not work with right censored survival data, right?

Furthermore - apart from comparing different survival model algorithms - how can I show that my new Random Survival Forest model can predict patient's survival better than other common prognosis markers (i.e. tumor markers, tumor classifications, already published scores, ...)? By plotting Feauture Importance plots?

I would be really glad if you could give me tips how to solve this issues with scikit-survival module!

Kind regards,
VanishingRasengan

**Comment by sebp** [comment, marked as answer]
It appears that there is no consensus, because the different measure tell you different things. Each has its benefits in certain situations. The c-index is certainly the most popular metric, but it only provides information about how well overall the model can discriminate between alive and dead (or whatever else your event denotes). Most importantly, the c-index doesn't tell you anything how well you are performing across different time horizons. That's exactly where the time-dependent ROC comes in. If you want to compare how well you can discriminate alive/dead after 1 year and 3 years, it can give you that information. The c-index cannot provide this information. Finally, the time-dependent Brier score assesses calibration too. That means, whether for a certain group of subjects the predicted probability of survival at time t matches what you observe in reality (our data). A model can discriminate alive/dead pretty well, but be bad at estimating calibrated probabilities. 

To show that a model performs better than some baseline, any of these measures could be used, but each will define "better" in a slightly better way. IMO, the time-dependent Brier score is the most comprehensive measure, because provides information about discrimination and calibration, whereas the other measures only evaluate discrimination. But in the end, it always depends on your particular problem, maybe you do not care about calibration.
